klarna-on-demand/src/main/java/com/klarna/ondemand/crypto/SharedPreferencesCryptoImpl.java

```diff
@@ -12,42 +11,19 @@
 import java.security.NoSuchAlgorithmException;
 import java.security.PrivateKey;
 import java.security.PublicKey;
-import java.security.Signature;
-import java.security.SignatureException;
 import java.security.spec.InvalidKeySpecException;
 import java.security.spec.PKCS8EncodedKeySpec;
 import java.security.spec.X509EncodedKeySpec;
 
-class CryptoSharedPreferencesBaseImpl implements Crypto {
+class SharedPreferencesCryptoImpl extends CryptoBase {
     private static final String PUBLIC_KEY = "PublicKey";
```

Comment:
Maybe make this a little more Klarna specific? It is not inconceivable for the hosting app to also store a "PublicKey" or "PrivateKey" value here.